sdk/lib/core/iterable.dart

```diff
@@ -369,6 +369,8 @@ abstract mixin class Iterable<E> {
   /// final result = numbers.reduce((value, element) => value + element);
   /// print(result); // 17.5
   /// ```
+  /// If the iterable has no elements it will throw a state error.
+  /// Consider using [fold] if the collection may be empty.
```

Comment:
We usually don't document errors thrown this way.
Instead we document requirements that must be satisfied, and then leave it as an implementation details what thappens if they're not.
Thet's what the "The iterable must have at least one element." sentence above says.

Adding a "Consider using \[fold\] if the iterable can be empty." line would reasonable.
(Using "iterable" instead of "collection" because not all iterables are backed by collections.)